internal/generator/vector/sources.go

```diff
@@ -33,14 +34,18 @@ func Sources(spec *logging.ClusterLogForwarderSpec, namespace string, op generat
 }
 
 func LogSources(spec *logging.ClusterLogForwarderSpec, namespace string, op generator.Options) []generator.Element {
-	var el []generator.Element = make([]generator.Element, 0)
+	var el = make([]generator.Element, 0)
 	types := generator.GatherSources(spec, op)
+	var outputTypes []string
+	for _, output := range spec.Outputs {
+		outputTypes = append(outputTypes, output.Type)
```

Comment:
Should this be name instead of type?  The exclusion method compares well known name to build the paths

Reply:
name can be any string, but type is described and controlled by API
https://github.com/openshift/cluster-logging-operator/blob/master/apis/logging/v1/cluster_log_forwarder_types.go#L151

Reply:
these are well known names by us; they will not change.  Loki, is the same unless we set something specific in cl.logStore

Reply:
maybe we talk about different things by name i mean this  https://github.com/openshift/cluster-logging-operator/blob/master/internal/generator/vector/conf_test.go#L435

Reply:
This confirms what I was asking: https://github.com/openshift/cluster-logging-operator/pull/1757/files#diff-e72ab332759bfa5c65f11425dfa95625f13ec0b617f5c4d3c23151c6ea0445f0R110-R113  It just so happens we chose an output type to be the same as the name (e.g. "Elasticsearch"), but we need to exclude by name an not type.  Instead of conditionally adding Loki, I would just add it to the list along with the others.  I do not think users can deploy Loki to openshift-logging and use it as the default using another name.  I'll ask @xperimental or @periklis to comment

Reply:
@vparfonov conversation with @periklis suggested excluding all of `openshift-logging`.  This would simplify the change.  Need to be aware of we want to include the eventrouter so we will need to make sure we get the include/exclude correctly

Reply:
@vimalk78 does vector do exclusions by pod label? instead of path?

Reply:
@jcantrill https://vector.dev/docs/reference/configuration/sources/kubernetes_logs/#container-exclusion